src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao.impl;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement createBookStatement = connection.prepareStatement(query,
+                     Statement.RETURN_GENERATED_KEYS)) {
+            createBookStatement.setString(1, book.getTitle());
+            createBookStatement.setBigDecimal(2, book.getPrice());
+            createBookStatement.executeUpdate();
+            ResultSet resultSet = createBookStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getObject(1, Long.class));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create book "
+                    + book + ". ", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT id, title, price "
+                + "FROM books "
+                + "WHERE id = ?;";
+        Book book = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = getBook(resultSet);
+            }
+            return Optional.ofNullable(book);
```

Comment:
Minor remark. You can do smth like that:
if (resultSet.next()) {
    return Optional.of(getBook(resultSet);
}
and instead of this Optional.ofNullable(book);, return just Optional.empty()